src/USER-MISC/pair_agni.cpp

```diff
@@ -347,132 +358,123 @@ double PairAGNI::init_one(int i, int j)
       memset(params,0,nparams*sizeof(Param));
+      
       curparam = -1;
-
-    } else if (params && (nwords == nparams+1) && (strcmp(words[0],"element") == 0)) {
       wantdata = -1;
+
+      values = reader.next_values(2);
+      values.next_string(); // ignore
       for (i = 0; i < nparams; ++i) {
+        std::string element = values.next_string();
         for (j = 0; j < nelements; ++j)
-          if (strcmp(words[i+1],elements[j]) == 0) break;
+          if (strcmp(element.c_str(),elements[j]) == 0) break;
         if (j == nelements)
           error->all(FLERR,"No suitable parameters for requested element found");
         else params[i].ielement = j;
       }
-    } else if (params && (nwords == 2) && (strcmp(words[0],"interaction") == 0)) {
-      for (i = 0; i < nparams; ++i)
-        if (strcmp(words[1],elements[params[i].ielement]) == 0) curparam = i;
-    } else if ((curparam >=0) && (nwords == 1) && (strcmp(words[0],"endVar") == 0)) {
-      int numtrain = params[curparam].numtrain;
-      int numeta = params[curparam].numeta;
-      params[curparam].alpha = new double[numtrain];
-      params[curparam].yU = new double[numtrain];
-      params[curparam].xU = new double*[numeta];
-      for (i = 0; i < numeta; ++i)
-        params[curparam].xU[i] = new double[numtrain];
 
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      for (i = 0; i < nparams; ++i){
+        std::string element = values.next_string();
+        if (strcmp(element.c_str(),elements[params[i].ielement]) == 0) curparam = i;
+      }
+
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      atomic_feature_version = values.next_int();
+      if (atomic_feature_version != AGNI_VERSION_1 && atomic_feature_version != AGNI_VERSION_2)
+        error->all(FLERR,"Incompatible AGNI potential file version");
+
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      params[curparam].numeta = values.next_int();
+      params[curparam].eta = new double[params[curparam].numeta];
+      params[curparam].xU = new double*[params[curparam].numeta];
+
+      values = reader.next_values(params[curparam].numeta + 1);
+      values.next_string(); // ignore
+      for(i = 0; i < params[curparam].numeta; i++)
+        params[curparam].eta[i] = values.next_double();
+      
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      params[curparam].gwidth = values.next_double();
+      
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      params[curparam].cut = values.next_double();
+      
+      values = reader.next_values(2);
+      values.next_string(); // ignore
+      params[curparam].numtrain = values.next_int();
+      params[curparam].alpha = new double[params[curparam].numtrain];
+      for (i = 0; i < params[curparam].numeta; ++i)
+        params[curparam].xU[i] = new double[params[curparam].numtrain];
```

Comment:
Here are more allocations that need to be repeated and data communicated for MPI rank != 0.